### DISCUSSION TITLE
Custom tsconfig.json location

### DISCUSSION THREAD

**Discussion by jeppe-smith** [remix-run/remix#3277] [Proposals]
It would be nice if we could specify a custom location for ´tsconfig.json´ in `remix.config.js`. Something like:

```js
module.exports = {
  ignoredRouteFiles: ['**/.*'],
  tsconfigPath: "./config/tsconfig.remix.json"
};
```

Since `"noEmit": true` is required for remix it makes it difficult to use remix on an express server written in typescript. Unless I'm missing something which is entirely possible.

**Comment by MoSattler** [comment]
Yes, it's a big problem for me. Any chance this could be implemented?

**Comment by arielvieira** [comment]
I've been using this patch for a couple months and had no issues
```diff
diff --git a/node_modules/@remix-run/dev/dist/config.js b/node_modules/@remix-run/dev/dist/config.js
index 1a16c23..5b891e0 100644
--- a/node_modules/@remix-run/dev/dist/config.js
+++ b/node_modules/@remix-run/dev/dist/config.js
@@ -259,7 +259,9 @@ async function readConfig(remixRoot, serverMode = serverModes.ServerMode.Product
   let tsconfigPath;
   let rootTsconfig = path__default["default"].resolve(rootDirectory, "tsconfig.json");
   let rootJsConfig = path__default["default"].resolve(rootDirectory, "jsconfig.json");
-  if (fse__default["default"].existsSync(rootTsconfig)) {
+  if (appConfig.tsconfigPath) {
+    tsconfigPath = appConfig.tsconfigPath
+  } else if (fse__default["default"].existsSync(rootTsconfig)) {
     tsconfigPath = rootTsconfig;
   } else if (fse__default["default"].existsSync(rootJsConfig)) {
     tsconfigPath = rootJsConfig;
```
Here's Remix source file for Reference: https://github.com/remix-run/remix/blob/main/packages/remix-dev/config.ts#L730


Any chance we could get this into Remix? it seams an easy feature to add. Would be really awesome having this functionality exposed to the `AppConfig`.

**Comment by arielvieira** [comment]
Update: there's a open PR that would enable this feature here https://github.com/remix-run/remix/pull/6301
